### PR TITLE
Allow adding /health endpoint to OpenAPI using config

### LINKED ISSUE #11965: Config property to include health endpoints in OpenAPI definition
**Description**
As [discussed in Zulip](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/openapi.20definition.20for.20.2Fhealth.20endpoints) it could be useful to include the health endpoints (like `/health`) in the OpenAPI definition for the application.

Our use case for this is that we are deploying our applications to Azure and register all endpoints with its API management. If an endpoint (like `/health`) isn't registered there, it can't be called by other applications. For this reason we would like to have an option to include the MicroProfile Health endpoints in the generated OpenAPI definition.

**Implementation ideas**
As discussed on Zulip one idea would be to add a new configuration property to the `quarkus-smallrye-openapi` extension, which when set and `quarkus-smallrye-health` is available, would automatically add the health endpoints to the OpenAPI definition.

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix  #11965

Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

**Comment by kenfinnigan** [comment]
I'm wondering whether this approach needs adjusting to utilize a `BuildItem` to capture additional OpenAPI schema bits, which the health extension can then produce?

Then how Health Checks are represented in OpenAPI document are defined within health extension.

**Comment by phillip-kruger** [comment]
Sounds reasonable. I'll change to do that. Then it's also easy for anyone to take part...

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/HealthOASFilter.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import io.smallrye.openapi.api.models.responses.APIResponseImpl;
+import io.smallrye.openapi.api.models.responses.APIResponsesImpl;
+
+/**
+ * This adds /health endpoint to the document.
+ */
+public class HealthOASFilter implements OASFilter {
+    private static final String ROOT_PATH = "quarkus.smallrye-health.root-path";
+    private static final String LIVENESS_PATH = "quarkus.smallrye-health.liveness-path";
+    private static final String READINESS_PATH = "quarkus.smallrye-health.readiness-path";
```
Comment (machi1990) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/HealthOASFilter.java:37]: Suggestion: maybe we can retrieve the values represented by these configuration through the config root classes:
 https://github.com/quarkusio/quarkus/blob/e76d50703dbe5e116a3ce9ce7b57e65c2ca8f017/extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthConfig.java#L8

We might need to create a spi module for that.

File: extensions/smallrye-openapi-common/deployment/src/main/java/io/quarkus/smallrye/openapi/common/deployment/SmallRyeOpenApiConfig.java [hunk 2]
```diff
@@ -20,4 +20,11 @@
      */
     @ConfigItem
     public Optional<Path> storeSchemaDirectory;
+
+    /**
+     * This allows you to add the /heath endpoints (from MicroProfile Health) to the
```
Comment (gastaldi) [inline extensions/smallrye-openapi-common/deployment/src/main/java/io/quarkus/smallrye/openapi/common/deployment/SmallRyeOpenApiConfig.java:25]:
```suggestion
     * This allows you to add the Liveness and Readiness Health endpoints (from MicroProfile Health) to the
```

**Comment by phillip-kruger** [comment]
Closing here. Supersede by #12044

**Pull request closed without merging**